### PR TITLE
DRILL-7324: Final set of "batch count" fixes

### PR DESCRIPTION
**Pull request by paul-rogers**
Final set of fixes for batch count/record count issues. Enables
vector checking for all operators (except statistics, which do not
produce valid Drill batches.)

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java [hunk 1]
```diff
@@ -82,17 +83,27 @@
  *       "sales_city" : BIGINT - nonnullstatcount(sales_city)
  *       "cnt"        : BIGINT - nonnullstatcount(cnt)
  *   .... another map for next stats function ....
+ * </pre>
+ * <p>
+ * Note that the above schema is not a valid Drill record batch: the varous
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java]:
```suggestion
 * Note that the above schema is not a valid Drill record batch: the various
```

Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java]: @vvysotskyi could you please take a look at this java doc comment and changes in this class?

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java [hunk 2]
```diff
@@ -226,7 +226,8 @@ public int next() {
           }
           ++parseErrorCount;
           if (printSkippedMalformedJSONRecordLineNumber) {
-            logger.debug("Error parsing JSON in " + hadoopPath.getName() + " : line nos :" + (recordCount + parseErrorCount));
+            logger.debug("Error parsing JSON in {}: line nos: {}",
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java]: What `nos` means?

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java [hunk 3]
```diff
@@ -256,6 +257,7 @@ private void updateRunningCount() {
   public void close() throws Exception {
     if(stream != null) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java]:
```suggestion
    if (stream != null) {
```

File: exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java [hunk 4]
```diff
@@ -81,6 +81,8 @@ public void testEmptyList() throws Exception {
 
   @Test
   public void schemaChange() throws Exception {
+    // TODO: What is this testing? Presumably, we are simply
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java]: I assume yes, it just tests that test pass, do we need this comment here though?

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java (hunk 1 above)
Comment (vvysotskyi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java]: If I understand these changes from DRILL-1328 correctly, as a result, we should obtain a map with a single value and its children also should have single values. Looks like this is a bug with specifying wrong row count (perhaps in `buildOutgoingRecordBatch()` method).

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java (hunk 2 above)
Comment (paul-rogers) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java]: I suspect author means "numbers", but of course there is only one. Changed to just "line:".

File: exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java (hunk 4 above)
Comment (paul-rogers) [reply exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonReader.java]: Revised the comment.

**Comment by arina-ielchiieva** [comment]
+1, LGTM

**Pull request closed without merging**